services/index/txindex.go

```diff
@@ -466,7 +466,8 @@ func (idx *TxIndex) ConnectBlock(dbTx database.Tx, block *types.SerializedBlock,
 	if node == nil {
 		return fmt.Errorf("no node %s", block.Hash())
 	}
-	if !node.GetStatus().KnownInvalid() {
+
+	if idx.chain.CacheInvalidTx || !node.GetStatus().KnownInvalid() {
```

Comment:
1.) IMO, the invalidTx should not share the same index DB with the validTx. The Tx index DB is built by default since it is required. the invalid index DB should not build by default. It should build separately and only if the user asked to do so and specified using `--debugtxindex` flag ( the `--cacheinvalidtx` flag name might raise the question why we cache invalid tx?)
2.) Or You think it is good enough to use the same index DB for all tx whether it's valid or invalid. Because its valid status depends only on the blocks' status on the BlockDAG node. 
3.) even If 2.) is true. I think you should still need to consider the inconsistent status if the user first starts the node by using the default config(invalid-no-indexed). then user stops & and restart with invalid-indexed. and then restarts to switch back to invalid-no-indexed. Then we got an inconsistent status. some invalid tx can access by RPC with invalid-no-indexed configuration which might be indexed in the user step 2.